.github/workflows/ci.yml

```diff
@@ -1,91 +1,112 @@
 name: CI
 
 on:
-  push:
   pull_request:
-  schedule: # Trigger a job on master at midnight UTC every day
-    - cron: '0 0 * * *'
-
+  push:
+    branches:
+      - ros2
+      - master
+      - ci-dev
+ 
 jobs:
+  # amd64_build job by default runs the following in additon to building the ros2 packages:
+  #  - copyright
+  #  - cppcheck
+  #  - cpplint
+  #  - lint_cmake
+  #  - linter
+  #  - xmllint
+  #  - uncrustify
   amd64_build:
+    runs-on: ubuntu-20.04
     strategy:
-      matrix:
-        env:
-          - {ROS_DISTRO: melodic, ROS_REPO: main, CMAKE_ARGS: "-Wall -Wextra -Wpedantic -Werror", TARGET_WORKSPACE: "$TARGET_REPO_PATH metapackage_dependencies.rosinstall", UPSTREAM_WORKSPACE: "upstream_dependencies.rosinstall"}
-          - {ROS_DISTRO: melodic, ROS_REPO: testing, CMAKE_ARGS: "-Wall -Wextra -Wpedantic -Werror", TARGET_WORKSPACE: "$TARGET_REPO_PATH metapackage_dependencies.rosinstall", UPSTREAM_WORKSPACE: "upstream_dependencies.rosinstall"}
-
-    runs-on: ubuntu-18.04
+      fail-fast: false
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Run ROS Industrial CI
-        uses: ros-industrial/industrial_ci@master
-        env: ${{matrix.env}}
-
-  clang-format:
+    - name: Set Environment Variables 
+      run: |
+        echo "name=LANG" >> $GITHUB_ENV
+
+    - name: Setup ROS 2
+      uses: ros-tooling/setup-ros@0.1.4
+      with:
+        required-ros-distributions: foxy
+    
+    - name: Run Tests
```

Comment:
you should probably run `ros-tooling/action-ros-lint` here before building so that it doesnt bother to waste build resources on this job if it fails the lint checks.